File: 3174-MinimumNumberOfChangesToMakeBinaryStringBeautiful/3174-MinimumNumberOfChangesToMakeBinaryStringBeautiful.java
// Last updated: 7/1/2025, 11:03:33 PM
class Solution {
    public int minChanges(String s) {
        int index = 0 ; 
        int count = 0; 

        while(index < s.length() -1 ){
            if(s.charAt(index) != s.charAt(index + 1)){
                count++; 
            }
            index += 2; 
        }

        return count; 
    }
}

File: 3174-MinimumNumberOfChangesToMakeBinaryStringBeautiful/3174-MinimumNumberOfChangesToMakeBinaryStringBeautiful.py
# Last updated: 7/1/2025, 11:03:43 PM
class Solution:
    def minChanges(self, s: str) -> int:
        index = 0 
        count = 0; 


        while (index < len(s) -1):
            if(s[index + 1] != s[index]):
                count +=1
            index += 2

        return count; 